pkg/cloud/services/ec2/instances.go

```diff
@@ -1154,6 +1154,10 @@ func getInstanceMarketOptionsRequest(i *infrav1.Instance) (*ec2.InstanceMarketOp
 		return nil, errors.New("can't create spot capacity-blocks, remove spot market request")
 	}
 
+	if (i.MarketType == infrav1.MarketTypeSpot || i.SpotMarketOptions != nil) && i.CapacityReservationID != nil {
+		return nil, errors.New("can't create spot capacity-blocks, remove capacityReservationID or remove spot requests")
```

Comment:
```suggestion
		return nil, errors.New("unable to generate marketOptions for spot instance, capacityReservationID is incompatible with marketType spot and spotMarketOptions")
```

Reply:
Could you also please change the `		return nil, errors.New("can't create spot capacity-blocks, remove spot market request")` above at line 1158 to a similarly formatted error?
Thanks!